### ISSUE TITLE
After Startup Error, adb again am start Arknights

### ISSUE THREAD

**Issue by Constrat** [MaaAssistantArknights/MaaAssistantArknights#8959]
### The problems you have encountered?

This seems to mainly an Overseas issue (perhaps only Global).
What happens is that sometimes the `am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext` will run and will start Arknights correctly, but because the game has an awful codebase, it will instantly crash on the emulator. 
MAA will think everything is going accordingly and will start running its tasks leading to this:

<details>
<summary>gui.log</summary>

```py
[2024-05-03 07:00:38.297] <1><> Start task: StartUp
[2024-05-03 07:00:46.502] <1><> Screencap takes a long time (avg: 608ms) which may cause some abnormal behaviors using automatic combat functions (e.g. Auto I. S.).
[2024-05-03 07:02:08.988] <1><> Task error: StartUp
[2024-05-03 07:02:09.491] <1><> Start task: Infrast
[2024-05-03 07:03:54.976] <1><> preferred operators: 斥罪
[2024-05-03 07:03:57.843] <1><> preferred operators: 斥罪
[2024-05-03 07:04:00.885] <1><> preferred operators: 斥罪
[2024-05-03 07:04:04.011] <1><> preferred operators: 斥罪
[2024-05-03 07:06:37.932] <1><> Complete task: Infrast
[2024-05-03 07:06:38.430] <1><> Start task: Mall
[2024-05-03 07:07:25.977] <1><> Task error: Mall
[2024-05-03 07:07:26.477] <1><> Start task: Recruit
[2024-05-03 07:07:49.156] <1><> Error occurred，Has returned
[2024-05-03 07:07:49.165] <1><> Task error: Recruit
[2024-05-03 07:07:49.657] <1><> Start task: Fight
[2024-05-03 07:07:56.423] <1><> Task error: Fight
[2024-05-03 07:07:56.923] <1><> Start task: Award
[2024-05-03 07:08:21.595] <1><> Task error: Award
[2024-05-03 07:08:21.622] <1><> All task(s) completed!
```

</details>

I've been trying different solutions for the past year such as:
- Switching Emulators: 
	- Mumu
	- Bluestacks
	- Memu
	
	They alle have a 1/10 chance of not starting the game correctly. (After a failed try usually the next one will also fail, from my testing Mumu has been the worse one, sad considering it has more performance out of all of them)
- Using different Arknights installation methods: 
	- directly from Google Play Store
	- .xapk
	- manually installing .apk + extracting .obb from the official Arknights Discord

- Tweaking instance settings such as:
	- Changing library support (Only ABI-32 as suggested for Arknights)
	- Reducing / Increasing core count
	- Changing the renderer and the engine

- Changing how MAA starts Arknights. For the past week I've been trying to leave Arknights open in the background and just returning home after finishing tasks, in the hope of skipping the startup process for the emulator (which I thought was the cause), but apparently after X amount of time Arknights will need a restart, and if that happens + the issue I'm describing, the rotation will be completely lost. So even this try was to no solution.

### Any good ideas?

Unfortunately, there's not much that can be done, emulator side, as we are just operating through ADB.
Plus I'm also running a completely automated CN account and the bilibili Arknights version NEVER and I say NEVER gives any type of startup issue, so I'm really thanking Yostar for this one.
I was hoping that in the case of a `StartUp TaskChainError`, instead of skipping to the next available task, the adb command `am start -n com.YoStarEN.Arknights/com.u8.sdk.U8UnityContext` could be ran one more time, repeating the StartUp task, hoping that this time the StartUp finishes correctly.
```py
[2024-05-03 07:02:08.988][INF][Px2a58][Tx16c4] Assistant::append_callback | SubTaskError {"class":"asst::ProcessTask","details":{},"first":["StartAtHome","StartWithSanity","StartUpBegin"],"pre_task":"StartUpBegin","subtask":"ProcessTask","taskchain":"StartUp","taskid":25,"uuid":"5c67b16a8af5066f"}
[2024-05-03 07:02:08.988][TRC][Px2a58][Tx16c4] asst::ProcessTask::on_run_fails | enter
[2024-05-03 07:02:08.988][TRC][Px2a58][Tx16c4] asst::ProcessTask::on_run_fails | leave, 0 ms
[2024-05-03 07:02:08.988][INF][Px2a58][Tx16c4] Assistant::append_callback | TaskChainError {"taskchain":"StartUp","taskid":25,"uuid":"5c67b16a8af5066f"}
```

This might seem very stupid, but even just spamming in the adb shell the start package, seems to make it work.
I've been doing some tests, on both Mumu and Bluestacks. Apparently spamming it about 5 times, with a 1 second delay seems to reduce the issue a lot.

So instead of checking for a StartUp Error, which could take up to 2 minutes. An option to spam the am start command, instead of only once.
During my testing spamming the command does not seem to create any unwanted exceptions, because if the game is running and is the foremost application, nothing will happen and the game will keep non running without any issue.
`Warning: Activity not started, intent has been delivered to currently running top-most instance.`

### Additional

I understand this is not a priority, as the CN client does not suffer from this bullshit, so maybe this could be introduced as a possible toggle?
I would add it myself, but I'm having troubles interfacing to adb, from cpp.

Thank you for your time.

<hr>

EDIT: After a bit more testing on Mumu, it seems the max number of "not starts" seems to be 3.
First 3 times it won't start, the 4th am start will keep the application running.
So the adb command could be started just 4 times, with a 2 second gap.

EDIT2: This is crazy, this time it needed 8 starts! It kept on crashing.
